go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -103,6 +110,16 @@ func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query string, callback func(*sq
 			session.Options.ClientFoundRows = true
 		}
 	}
+
+	if !session.InTransaction {
+		atomic.AddInt32(&busyConnections, 1)
+	}
+	defer func() {
+		if !session.InTransaction {
+			atomic.AddInt32(&busyConnections, -1)
+		}
+	}()
```

Comment:
I did this semi-complicated logic so that we would consider connections as active during the life of a transaction, but on second thought, maybe we should just allow the vtgate to terminate mid-transaction and ensure that the rollback occurs?